apps/aecore/src/aec_tx_pool.erl

```diff
@@ -348,67 +354,70 @@ int_get_candidate(MaxGas, BlockHash, #dbs{db = Db} = DBs) ->
                        _ATrees, _Height, Acc) when Gas =< GasLimit ->
     {ok, Gas, Acc};
-int_get_candidate_fold(Gas, GasLimit, Db, Dbs = #dbs{}, {Txs, Cont},
+int_get_candidate_fold(Gas, GasLimit, MinMinerGasPrice, Db, Dbs = #dbs{}, {Txs, Cont},
                        AccountsTree, Height, Acc) when Gas > GasLimit ->
-    {RemGas, NewAcc} = fold_txs(Txs, Gas, GasLimit, Db, Dbs,
+    {RemGas, NewAcc} = fold_txs(Txs, Gas, GasLimit, MinMinerGasPrice, Db, Dbs,
                                 AccountsTree, Height, Acc),
-    int_get_candidate_fold(RemGas, GasLimit, Db, Dbs, ets:select(Cont),
+    int_get_candidate_fold(RemGas, GasLimit, MinMinerGasPrice, Db, Dbs, ets:select(Cont),
                            AccountsTree, Height, NewAcc);
-int_get_candidate_fold(RemGas, _GL, _Db, _Dbs, '$end_of_table', _AccountsTree,
+int_get_candidate_fold(RemGas, _GL, _MMGP, _Db, _Dbs, '$end_of_table', _AccountsTree,
                        _Height, Acc) ->
     {ok, RemGas, Acc}.
 
-fold_txs([Tx|Txs], Gas, GasLimit, Db, Dbs, AccountsTree, Height, Acc) ->
+fold_txs([Tx|Txs], Gas, GasLimit, MinMinerGasPrice, Db, Dbs, AccountsTree, Height, Acc) ->
     if Gas > GasLimit ->
             {Gas1, Acc1} =
                 int_get_candidate_(
-                  Tx, Gas, Db, Dbs, AccountsTree, Height, Acc),
-            fold_txs(Txs, Gas1, GasLimit, Db, Dbs, AccountsTree, Height, Acc1);
+                  Tx, Gas, MinMinerGasPrice, Db, Dbs, AccountsTree, Height, Acc),
+            fold_txs(Txs, Gas1, GasLimit, MinMinerGasPrice, Db, Dbs, AccountsTree, Height, Acc1);
        true ->
             {Gas, Acc}
     end;
-fold_txs([], Gas, _, _, _, _, _, Acc) ->
+fold_txs([], Gas, _, _, _, _, _, _, Acc) ->
     {Gas, Acc}.
 
 int_get_candidate_({?KEY(_, _, Account, Nonce, _) = Key, Tx},
-                   Gas, Db, Dbs, AccountsTree, Height, Acc) ->
+                   Gas, MinMinerGasPrice, Db, Dbs, AccountsTree, Height, Acc) ->
     case gb_trees:is_defined({Account, Nonce}, Acc) of
         true ->
             %% The earlier must have had higher fee. Skip this tx.
             {Gas, Acc};
         false ->
             check_candidate(
-              Db, Dbs, Key, Tx, AccountsTree, Height, Gas, Acc)
+              Db, Dbs, Key, Tx, AccountsTree, Height, Gas, MinMinerGasPrice, Acc)
     end.
 
-
 check_candidate(Db, #dbs{gc_db = GCDb} = Dbs,
                 ?KEY(_, _, Account, Nonce, TxHash) = Key, Tx,
-                AccountsTree, Height, Gas, Acc) ->
+                AccountsTree, Height, Gas, MinMinerGasPrice, Acc) ->
     Tx1 = aetx_sign:tx(Tx),
     TxTTL = aetx:ttl(Tx1),
     TxGas = aetx:gas_limit(Tx1, Height),
-    case Height < TxTTL andalso TxGas > 0 andalso ok =:= int_check_nonce(Tx, AccountsTree, check_candidate) of
+    case Height < TxTTL andalso TxGas > 0
+         andalso ok =:= int_check_nonce(Tx, AccountsTree, check_candidate)
+         andalso MinMinerGasPrice =< aetx:min_gas_price(Tx1, Height) of
```

Comment:
I see aetx:min_gas_price uses aetx:gas_limit as denominator, and aetx:gas_limit returns zero as error, though this is safe here because `TxGas > 0` present in andalso close preceding. Ok.